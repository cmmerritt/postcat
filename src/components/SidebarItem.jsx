import React from 'react';
import PropTypes from 'prop-types';

const SidebarItem = ({ method, urlText }) => (
  <section>
    <h4>{method}</h4>
    <span>{urlText}</span>
  </section>
);

SidebarItem.propTypes = {
  method: PropTypes.string.isRequired, 
  urlText: PropTypes.string.isRequired
};

export default SidebarItem;



