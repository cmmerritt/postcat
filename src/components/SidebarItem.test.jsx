import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SidebarItem from './SidebarItem.jsx';

describe('SidebarItem component', () => {
  afterEach(() => cleanup());
  it('renders SidebarItem', () => {
    const { asFragment } = render(<SidebarItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});
