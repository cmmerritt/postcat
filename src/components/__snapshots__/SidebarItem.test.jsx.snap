// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem component renders SidebarItem 1`] = `
<DocumentFragment>
  <section
    class="sidebarItem"
  >
    <h4 />
    <span />
  </section>
</DocumentFragment>
`;
